lib/spack/spack/filesystem_view.py

```diff
@@ -41,15 +41,26 @@
 _projections_path = '.spack/projections.yaml'
 
 
-def view_symlink(src, dst, view, spec):
+def view_symlink(src, dst, **kwargs):
+    # keyword arguments are irrelevant
+    # here to fit required call signature
     os.symlink(src, dst)
 
 
-def view_hardlink(src, dst, view, spec):
+def view_hardlink(src, dst, **kwargs):
+    # keyword arguments are irrelevant
+    # here to fit required call signature
     os.link(src, dst)
 
 
-def view_copy(src, dst, view, spec):
+def view_copy(src, dst, **kwargs):
+    """
+    Copy a file from src to dst.
+
+    Use spec and view from kwargs to generate relocations
+    """
+    spec = kwargs.get(spec, None)
+    view = kwargs.get(view)  # functools assures this is present
```

Comment:
```suggestion
def view_copy(src, dst, view, spec=None):
    """
    Copy a file from src to dst.

    Use spec and view to generate relocations
    """
```

Reply:
I thought I had tried that and it didn't work, but I tried it again and it seems to be working... hmm